### PR TITLE
kubeadm: remove code to be removed in 1.30

### PR DESCRIPTION
**Pull request by HirazawaUi**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
According to the comments, we don't need this code in 1.30
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
part of [#2414](https://github.com/kubernetes/kubeadm/issues/2414)

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://git.k8s.io/enhancements/keps/sig-cluster-lifecycle/kubeadm/4214-separate-super-user-kubeconfig
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
thanks,

how about removing the TODOs from this PR as well?
https://github.com/kubernetes/kubernetes/pull/121841/files

you can add two commits in this PR

it's related to the same KEP, even if it's not mentioned in the KEP doc:
> https://git.k8s.io/enhancements/keps/sig-cluster-lifecycle/kubeadm/4214-separate-super-user-kubeconfig

**Comment by HirazawaUi** [comment]
> how about removing the TODOs from this PR as well? https://github.com/kubernetes/kubernetes/pull/121841/files
> 
> you can add two commits in this PR

A new commit has been submitted, please review whether it is correct.

File: cmd/kubeadm/app/phases/certs/renewal/manager.go [hunk 1]
```diff
@@ -421,50 +400,3 @@ func certToConfig(cert *x509.Certificate) certutil.Config {
...
-		kubeadmconstants.APIServerEtcdClientCertAndKeyBaseName:
-		return []certConfigMutatorFunc{
-			removeSystemPrivilegedGroupMutator(),
-		}
-	case kubeadmconstants.APIServerKubeletClientCertAndKeyBaseName:
-		return []certConfigMutatorFunc{
-			removeSystemPrivilegedGroupMutator(),
-			addClusterAdminsGroupMutator(),
-		}
-	}
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/certs/renewal/manager.go:439]: let's remove only this code that i selected above and the two functions removeSystemPrivilegedGroupMutator addClusterAdminsGroupMutator.

keep the rest of the mutator code, in the future we might need it.

Comment (SataQiu) [reply cmd/kubeadm/app/phases/certs/renewal/manager.go:439]: +1

Comment (HirazawaUi) [reply cmd/kubeadm/app/phases/certs/renewal/manager.go:439]: all right

**Review by neolit123 — COMMENTED** [review]
/approve
/cc @SataQiu 
PTAL for LGTM

**Review by neolit123 — COMMENTED** [review]
/lgtm
/triage accepted

**Pull request merged**